### ISSUE TITLE
[Bug]: Password Autofill Provider shows on termux

### ISSUE THREAD

**Issue by mio-19** [termux/termux-app#3909]
### Problem description

Password Autofill Service shows on termux terminal, which can disturb with the workflow.

![Screenshot_20240407_020702_Termux](https://github.com/termux/termux-app/assets/57285379/6502e0cb-3a92-4547-9cdf-d8d096d71ac2)

### Steps to reproduce the behavior.

1. Install Bitwarden and set it as the auto-fill provider. 
2. Open Termux

### What is the expected behavior?

No Autofill

### System information

* Termux application version: 0.118.0
* Android OS version: 14
* Device model: Samsung S23 Ultra

**Comment by Biswa96** [comment]
That autofill option is provided by bitwarden and termux is not related with that. You have to disable it from bitwarden or Android settings.

**Comment by fwrs** [comment]
Also experiencing this with 1Password where it is a little more obtrusive. I don't think this is 1Password's fault though, I've not had it show up on non-password text fields in any other app. It's likely that there's an existing reason why Termux does it this way, so I'd appreciate it if this were to be a configurable setting.

![Screenshot_20240409_005952_Termux](https://github.com/termux/termux-app/assets/44343874/acc23ff1-ba74-4ce0-a41a-46a9a831c76b)

**Comment by yubrajbhoi** [comment]
Yeah, I also got this bug after a Gboard update. I just disabled the suggestion strip entirely after that. Here is how it looks like on my Google Pixel 6a:

![Screenshot_20240413-183536](https://github.com/termux/termux-app/assets/30350489/f1dbf5ba-511e-462e-a4e7-f210fdd4327d)

**Comment by juvannx** [comment]
> from bitwarden or Android settings.

Autofill is managed by Android so apps can't decide if they must show or no, in Android there is no settings for disable autofill for apps.
So termux should change components which accept input with android:autofillHints and disable it.

**Comment by juvannx** [comment]
> I just disabled the suggestion strip entirely after that. 

This is a workaround and not a solution.
Termux should update with new version of android.

**Comment by psyburr** [comment]
Since this was reported using bitwarden, I'd like to offer a solution to those with the same problem also using the bitwarden application. In the following Settings—>Autofill menu, at the very bottom there is an option to Block AutoFill. Selecting this will open a menu with a button labeled _New blocked URI_, selecting that will open a dialog where you can enter the following to prevent bitwarden from suggesting any auto-fill whenever using Termux. Screenshots from my device attached for visual reference. Hope this helps!

`androidapp://com.termux`

![https://i.imgur.com/ArqjFQZ.jpeg](url)

![https://i.imgur.com/PvHLt8W.jpeg](url)

**Comment by indrastorms** [comment]
> Since this was reported using bitwarden, I'd like to offer a solution to those with the same problem also using the bitwarden application. In the following Settings—>Autofill menu, at the very bottom there is an option to Block AutoFill. Selecting this will open a menu with a button labeled _New blocked URI_, selecting that will open a dialog where you can enter the following to prevent bitwarden from suggesting any auto-fill whenever using Termux. Screenshots from my device attached for visual reference. Hope this helps!
> 
> `androidapp://com.termux`
> 
Now you can't autofill when you wanted to.

**Comment by psyburr** [comment]
We never wanted to. Please see the initial report and expected behavior.

**Comment by indrastorms** [comment]
> We never wanted to. Please see the initial report and expected behavior.

So what do you think about the option "autofill password" existed for?

**Comment by yubrajbhoi** [comment]
@psyburr That is a pretty good workaround. Do you know if that is possible on stock Android (with no password manager)?

**Comment by yubrajbhoi** [comment]
All right, I found a workaround. Go to Settings and "Passwords, passkeys and autofill". Select the preferred service, for me on Pixel 6a, it is Google. That will take you to "Use Autofill with Google". Disable that and it should be fixed. If you are actually using Google password manager this won't work for you.

**Comment by escape0707** [comment]
@yubrajbhoi I've set it to Bitwarden since day one I bought my Pixel 6a. I don't know if switching it back to Google then again to Bitwarden will do anything... I'm using the "disable autofill for termux" workaround now.

**Comment by yubrajbhoi** [comment]
@escape0707 Yeah, it won't do anything. You have to disable the app you save set as preferred service. In my case I use `pass` in Termux as my password manager, so I don't need Google.

**Comment by Dhyfer1** [comment]
![Screenshot_20240826-093310.jpg](https://github.com/user-attachments/assets/5137636c-bd69-489c-9798-46d2c3bdc29b)

Please, anyone with any solution for Gboard and Google password manager? It's annoying to see this every time I open termux 😤

**Comment by agnostic-apollo** [comment]
Check #4125. Users using GitHub action builds should test and report so that it can be merged.

**Comment by Dhyfer1** [comment]
> Check #4125. Users using GitHub action builds should test and report so that it can be merged.

Thanks. In the last build action the annoying password button no longer appears above the keyboard. I hope the fix will come to the latest stable and beta version soon.

**Comment by agnostic-apollo** [comment]
Welcome. Should be available in next beta at least.

**Comment by indrastorms** [comment]
> Welcome. Should be available in next beta at least.

In that case the base branch of the pr should be 119.beta. right?

**Comment by agnostic-apollo** [comment]
Master branch is where everything gets pushed for final release, beta branches pick and patch commits from master branch before release.

**Comment by agnostic-apollo** [comment]
All the development is done on master branch, anything **I** push there is thoroughly tested, beta in the termux app's context means that features are not final and not ready for release and may change, not that they are unstable. Beta branches were never used before, it was only used for last release with specific commits picked from master branch and certain features disabled with patch commits that were not ready for release.

**Comment by indrastorms** [comment]
> All the development is done on master branch, anything **I** push there is thoroughly tested, beta in the termux app's context means that features are not final and not ready for release and may change, not that they are unstable. Beta branches were never used before, it was only used for last release with specific commits picked from master branch and certain features disabled with patch commits that were not ready for release.

Oh.. then it's more like a feat branch. But I'm using 119 so can't install 118 on top of it. 😞

**Comment by agnostic-apollo** [comment]
You can install github debuggable apks with `pm install -r -d` to downgrade.

**Comment by indrastorms** [comment]
Thanks for the info.

**Comment by h0tk3y** [comment]
As I use 1Password, I cannot filter the apps in the password manager, and I did not have the option to exclude Termux. 

The "Open 1Password" popup was very annoying to me. 

I managed to find another workaround: with Tasker, I disable (deselect) the autofill service in the system settings when Termux is focused and enable it back once another app becomes active.

The commands for ADB action in Tasker are: 

```bash
settings delete secure autofill_service
```

```bash
settings put secure autofill_service com.onepassword.android/com.onepassword.android.autofill.services.AutofillService
```

This works but requires Tasker to have access to ADB over TCP.

I hope the fix for this issue is included in a release soon so I can ditch my workaround.
